src/SuperMint.sol

```diff
@@ -24,10 +28,12 @@ contract SuperMint is IERC721Receiver {
     uint256,
     bytes memory
   ) public virtual override returns (bytes4) {
-    if (!complete) {
-      complete = true;
-      implementation.mint(maxMints - 1);
+    //check if iterator is greater than 1 and then reduce it's value
+    //before reentrance. When it's value drops below 1, reentrancy stops.
+    if (iterator > 1) {
+      iterator--;
+      implementation.mint(iterator);
     }
```

Comment:
for optimum efficiency, we should do:
```
if (maxMints - 1 != 0) {
  implementation.mint(maxMints);
  maxMints--;
}
```

maybe kinda counterintuitive but != 0 is cheaper than > or <